src/ds/d500/d500-types/calibration-config.h

```diff
@@ -0,0 +1,372 @@
+            {
+                if (!j.contains(field))
+                {
+                    throw std::invalid_argument(std::string("Invalid ROI format: missing field: ") + field);
+                }
+            }
+        }
+
+    private:
+        vertex m_vertex_0;
+        vertex m_vertex_1;
+        vertex m_vertex_2;
+        vertex m_vertex_3;
+    };
+
+    /*
+    camera_position class
+    handles JSON of representing the camera positon: rotation row major 3x3 matrix and translation vector
+    all values are floats.
+    JSON schema:
+    {
+        "rotation":
+            [
+                [a, b, c],
+                [d, e, f],
+                [g, h, i]
+            ],
+        "translation": {x, y, z}
+    }
+    */
+    class camera_position
+    {
+
+    public:
+        camera_position() = default;
+
+        camera_position(const json &j)
+        {
+            validate_json(j);
+            for (size_t i = 0; i < 3; ++i)
+            {
+                for (size_t k = 0; k < 3; ++k)
+                {
+                    m_rotation[i][k] = j.at("rotation")[i][k].get<float>();
+                }
+            }
+            for (size_t i = 0; i < 3; ++i)
+            {
+                m_translation[i] = j.at("translation")[i].get<float>();
+            }
+        }
+
+        json to_json() const
+        {
+            json j;
+            j["rotation"] = json::array();
+            for (size_t i = 0; i < 3; ++i)
+            {
+                j["rotation"].push_back(json::array());
+                for (size_t k = 0; k < 3; ++k)
+                {
+                    j["rotation"][i].push_back(m_rotation[i][k]);
+                }
+            }
+            j["translation"] = m_translation;
+            return j;
+        }
+
+    private:
+        void validate_json(const json &j) const
+        {
+            if (!j.is_object() || j.size() != 2 || !j.contains("rotation") || !j.contains("translation"))
+            {
+                throw std::invalid_argument("Invalid camera_position format: camera_position must include rotation and translation fields");
+            }
+            if (!j.at("rotation").is_array() || j.at("rotation").size() != 3)
+            {
+                throw std::invalid_argument("Invalid rotation format: rotation must be a 3x3 matrix");
+            }
+            for (size_t i = 0; i < 3; ++i)
+            {
+                if (!j.at("rotation")[i].is_array() || j.at("rotation")[i].size() != 3)
+                {
+                    throw std::invalid_argument("Invalid rotation row format: rotation must be a 3x3 matrix");
+                }
+                for (size_t k = 0; k < 3; ++k)
+                {
+                    if (!j.at("rotation")[i][k].is_number_float() && !j.at("rotation")[i][k].is_number_integer())
+                    {
+                        throw std::invalid_argument("Invalid rotation type: all rotation values must be floats.");
```

Comment:
You check float + int but the error message is just about int?

Reply:
Fixed. need to check only floats.